src/crypto/common.h

```diff
@@ -53,6 +53,13 @@ void static inline WriteLE64(unsigned char* ptr, uint64_t x)
     memcpy(ptr, (char*)&v, 8);
 }
 
+uint16_t static inline ReadBE16(const unsigned char* ptr)
+{
+    uint16_t x;
+    memcpy((char*)&x, ptr, 2);
+    return be16toh(x);
```

Comment:
Good, looks ilke we already have an implementation of `be16toh` in `src/compat/endian.h`, so it doesn't need to be added.